cpp/src/join/hash_join.cu

```diff
@@ -87,62 +94,86 @@ std::size_t compute_join_output_size(table_device_view build_table,
   }
 
   auto const probe_nulls = cudf::nullate::DYNAMIC{has_nulls};
-  pair_equality equality{probe_table, build_table, probe_nulls, nulls_equal};
 
-  row_hash hash_probe{probe_nulls, probe_table};
+  auto const row_hash           = cudf::experimental::row::hash::row_hasher{preprocessed_probe};
+  auto const hash_probe         = row_hash.device_hasher(probe_nulls);
   auto const empty_key_sentinel = hash_table.get_empty_key_sentinel();
-  make_pair_function pair_func{hash_probe, empty_key_sentinel};
+  auto const iter               = cudf::detail::make_counting_transform_iterator(
+    0, make_pair_function{hash_probe, empty_key_sentinel});
+
+  auto const row_comparator =
+    cudf::experimental::row::equality::two_table_comparator{preprocessed_probe, preprocessed_build};
 
```

Comment:
Better to remove this line.

Reply:
Which line is this exactly?

Reply:
Line 106. Sorry I can't select it since it is not new.